packages/rest-typings/src/v1/teams/index.ts

```diff
@@ -184,4 +186,8 @@ export type TeamsEndpoints = {
 			room: IRoom;
 		};
 	};
+
+	'/v1/teams.listChildren': {
+		GET: (params: TeamsListChildrenProps) => PaginatedResult<{ data: IRoom[] }>;
```

Comment:
more of a question here, since we're trying to be more flexible for extending this endpoint in the future, should we be more flexible here too? I mean having the response as `{ data: IRoom[] }` doesn't give much room for extending, if we want to add more data that is not an `IRoom` we would end up with something, like `{ data: IRoom[]; moreStuff: unknown[] }` .. 

so should we be more "specific" here, changing the response to something like:
```ts
{
  rooms: IRoom[];
  discussions: IRoom[];
}
```

I understand this will make the life of the frontend harder to mix the results, but not too much..

Reply:
Remember this endpoint is paginated, so separating the data in that way would also make pagination confusing. Assume we want a page of 50 elements fetching rooms/discussions. Should we:
- Return 50 of the first db call and 0 of the second
- Return 25 of both
If the first db call outputs just 1 element, then we would need to fetch 49 from the 2nd db call.
- Should the total of elements be the sum of both? or should we have separate totals (and separate paginations for each element)
Also, what happens with typings? If a user wants only rooms, should we return the `discussions` key as an empty array? 🤔 

So, i think that in the case we ever have a child of a team that's not an IRoom, we should modify this endpoint accordingly.

wdyt?

Reply:
good point.. so maybe adding a `type` property? something like:
```typescript
type RoomItem = { type: 'room'; data: IRoom };
type DiscussionItem = { type: 'discussion'; data: IRoom };
type NewFutureItem = { type: 'future'; data: IFuture };

type Response = {
  data: (RoomItem | DiscussionItem | NewFutureItem)[];
};
```
again, just thinking out loud =)

Reply:
🤔 that could work, but since the endpoint can return both rooms&discussions at the same time, then we would have the same problem. So not sure...